src/amo/components/SearchForm/index.js

```diff
@@ -114,20 +114,32 @@ export class SearchFormBase extends React.Component {
     this.setState({ searchValue: event.target.value });
   }
 
-  handleSuggestionsFetchRequested = this.props.debounce(({ value, reason }) => {
+  dispatchAutocompleteStart = this.props.debounce(({ filters }) => {
+    const { dispatch, errorHandler } = this.props;
+
+    dispatch(autocompleteStart({
+      errorHandlerId: errorHandler.id,
+      filters,
+    }));
+    this.setState({ oldAutocompleteFilters: filters });
+  }, 200, { leading: true })
```

Comment:
How does firing this off on the leading edge help? I don't think the first character that someone types would be useful.